32 bit compilation failed

## Version of Singularity:

Master and 3.3.0-rc3

### Expected behavior

Compile Singularity successfully

### Actual behavior

```
build flag -mod=vendor only valid when using modules
build flag -mod=vendor only valid when using modules
 GO singularity
    [+] GO_TAGS "containers_image_openpgp sylog imgbuild_engine oci_engine singularity_engine fakeroot_engine apparmor selinux seccomp"
 GEN etc/bash_completion.d/singularity
 GEN singularity.conf from /usr/local/etc/singularity/singularity.conf
 GO clean -cache
 GO cmd/starter/c/starter
# github.com/sylabs/singularity/internal/pkg/fakeroot
../internal/pkg/fakeroot/fakeroot.go:209:8: constant 4294901760 overflows int
Makefile:231: recipe for target 'cmd/starter/c/starter' failed
make: *** [cmd/starter/c/starter] Error 2
```

### Steps to reproduce behavior

Compile Singularity on a 32 bit system

--- a/internal/pkg/fakeroot/fakeroot.go
+++ b/internal/pkg/fakeroot/fakeroot.go
@@ -25,11 +25,11 @@ const (
 	// SubGIDFile is the default path to the subgid file.
 	SubGIDFile = "/etc/subgid"
 	// validRangeCount is the valid fakeroot range count.
-	validRangeCount = 65536
+	validRangeCount = uint32(65536)
 	// StartMax is the maximum possible range start.
-	startMax = 4294967296 - validRangeCount
+	startMax = uint32(4294967296 - 65536)
 	// StartMin is the minimum possible range start.
-	startMin = 65536
+	startMin = uint32(65536)
 	// disabledPrefix is the character prefix marking an entry as disabled.
 	disabledPrefix = '!'
 	// fieldSeparator is the character separating entry's fields.
@@ -207,7 +207,7 @@ func (c *Config) AddUser(username string) error {
 		return fmt.Errorf("could not retrieve user information for %s: %s", username, err)
 	}
 	for i := startMax; i >= startMin; i -= validRangeCount {
-		current := uint32(i)
+		current := i
 		available := true
 		for _, entry := range c.entries {
 			if entry.invalid {

--- a/internal/pkg/fakeroot/fakeroot_test.go
+++ b/internal/pkg/fakeroot/fakeroot_test.go
@@ -157,8 +157,8 @@ func createConfig(t *testing.T) string {
 
 	var buf bytes.Buffer
 
-	base := 10
-	size := 10
+	base := uint32(10)
+	size := uint32(10)
 
 	// valid users
 	for i := base; i < base+size; i++ {

